main.go

```diff
@@ -119,13 +116,15 @@ func main() {
 	setupLog.Info("Watching objects in namespace for reconciliation", "namespace", kubeNamespace)
 
 	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
-		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
-		HealthProbeBindAddress: probeAddr,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "e74dec1a.tinkerbell.org",
-		Namespace:              kubeNamespace,
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
+			BindAddress: metricsAddr,
+		},
+		// Port:                   9443,
+		HealthProbeBindAddress:  probeAddr,
+		LeaderElection:          enableLeaderElection,
+		LeaderElectionID:        "e74dec1a.tinkerbell.org",
+		LeaderElectionNamespace: kubeNamespace,
```

Comment:
Hey @chrisdoherty4, i could use your eyes here. I upgraded controller-runtime to v0.16.2 and didn't fully know how or if i correctly translated all the options over.

Reply:
The old `manager.Options.Namespace` is now `manager.Options.Cache.Namespaces`. We can modify the CLI params separately to allow specifying multiple.

We don't need to configure `LeaderElectionNamepsace`.

For the `Port` field we need to build a webhook server using `webhook.NewServer`. It accepts an options struct that we can configure the port on and then set `manager.Options.WebhookServer`.

Reply:
I'm a bit confused. Are we even deploying a webhook in Rufio?